File: src/main/java/fr/univ_smb/isc/m1/brutocollinus/infrastructure/persistence/entity/Boost.java
package fr.univ_smb.isc.m1.brutocollinus.infrastructure.persistence.entity;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToOne;

@Entity
public class Boost extends Nameable {
    @OneToOne(cascade = CascadeType.ALL)
    private FightStatistics fightStatistics;

    public Boost() {
        // JPA
    }

    public Boost(String name, FightStatistics fightStatistics) {
        super(name);
        this.fightStatistics = fightStatistics;
    }

    public FightStatistics fightStatistics() {
        return this.fightStatistics;
    }
}

File: frontend/src/components/Boost.js
import React from 'react'
import FightStatistics from './FightStatistics.js';

export default function Boost(props) {
    const boost = props.boost;

    return (
        <div>
            {boost.name}
            <FightStatistics fightStatistics={boost.fightStatistics}/>
        </div>
    );
}